src/ripple/app/tx/impl/Transactor.cpp

```diff
@@ -139,7 +139,20 @@ std::uint64_t Transactor::calculateBaseFee (
     if (tx.isFieldPresent (sfSigners))
         signerCount = tx.getFieldArray (sfSigners).size();
 
-    return baseFee + (signerCount * baseFee);
+    std::uint32_t memoCount = 0;
+
+    if (tx.isFieldPresent(sfMemos) && view.rules().enabled(featureMemoFee))
+    {
+        auto const& memos = tx.getFieldArray (sfMemos);
+
+        Serializer s (2048);
+        memos.add (s);
+
+        if (s.size() > 32)
+            memoCount = 2 * static_cast<std::uint32_t>(s.size());
```

Comment:
Seems reasonable. I'd probably consider making the multiplier a function of length too. For example:
`memoCount = std::max(1, (s.size() / 128)) * static_cast<std::uint32_t>(s.size());`

This gives it a nice "exponential-like" curve as you can see on this [WolframAlpha graph](https://www.wolframalpha.com/input/?i=plot+f(x)+%3D+max(1,+(x+%2F+128))+*+x+from+0+to+1024)

Reply:
Maybe just `s.size() * s.size() / 128` for simplicity. With a minimum `s.size()` of 32, this expression starts at 8 and scales to 8192.

Reply:
@miguelportilla makes a good point that super-linear fee scaling will just encourage splitting into multiple transactions at the point where the marginal cost of a memo byte exceeds the cost of a new transaction. That means we should just use `s.size()` for `memoCount`.